hack/functest.sh

```diff
@@ -2,10 +2,24 @@
 
 set -e
 
-# Override the image name when this is invoked from openshift ci
+# Override the image name, registry name when this is invoked from openshift ci
 if [ -n "$OPENSHIFT_BUILD_NAMESPACE" ]; then
-	CATALOG_FULL_IMAGE_NAME="registry.svc.ci.openshift.org/${OPENSHIFT_BUILD_NAMESPACE}/stable:cincinnati-operator"
-	echo "Openshift CI detected, deploying using image $CATALOG_FULL_IMAGE_NAME"
+	OPERATOR_IMAGE="registry.svc.ci.openshift.org/${OPENSHIFT_BUILD_NAMESPACE}/stable:cincinnati-operator"
+	GRAPH_DATA_IMAGE="registry.svc.ci.openshift.org/${OPENSHIFT_BUILD_NAMESPACE}/stable:cincinnati-graph-data-container"
+
+	echo "Openshift CI detected, deploying using image $OPERATOR_IMAGE and ${GRAPH_DATA_IMAGE}"
+
+	sed -i "s|quay.io/cincinnati/cincinnati-operator:latest|$OPERATOR_IMAGE|" deploy/operator.yaml
+	sed -i "s|quay.io/cincinnati/cincinnati:latest|quay.io/app-sre/cincinnati:2873c6b|" deploy/operator.yaml
+	sed -i "s|your-registry/your-repo/your-init-container|$GRAPH_DATA_IMAGE|" deploy/crds/cincinnati.openshift.io_v1beta1_cincinnati_cr.yaml
+
+else
+	if ! [ -n "$GRAPH_DATA_IMAGE" ]; then
+		echo "GRAPH_DATA_IMAGE environment variable must be set."
+		exit 1
+	fi
+	sed -i "s|quay.io/cincinnati/cincinnati:latest|quay.io/app-sre/cincinnati:2873c6b|" deploy/operator.yaml
+	sed -i "s|your-registry/your-repo/your-init-container|$GRAPH_DATA_IMAGE|" deploy/crds/cincinnati.openshift.io_v1beta1_cincinnati_cr.yaml
 fi
```

Comment:
Consider making two separate targets,`make deploy` and `make functest`.

Reply:
`make deploy` is useful outside of `make func-test`, when you just need the operator deployed.  And func-test would depend on deploy. Replacing values is an implementation detail on how the deploy is done.

Reply:
Ok, I can write a new target for deploy but then should creating a CR be not a part of the deploy? When we say to deploy the operator, I think it does mean creating a CR as well, please correct if I am wrong here. If I do not create the CR as a part of `make deploy` target then I think it is incomplete to say we have "deployed" the operator completely and then the purpose of that target is not fulfilled as then one will have to still manually create the CR. 

Regarding the (future) case if we have to create multiple CRs then the test case can be modified with BeforeEach and AfterEach at that time but as of now, I think I am covering all the functionalities we have in place currently. Please suggest if I am missing something that I am not testing for CR?

Reply:
Done.